checkPages.js

```diff
@@ -55,6 +55,36 @@ module.exports = function(host, options, done) {
     });
   }
 
+  /* Converts http://examplé.com/rosé?rosé=1 to
+   * http://examplé.com/ros%C3%A9?ros%C3%A9=1 so that servers which expect
+   * ASCII only requests can handle it */
+  function fixNonAsciiLink(link) {
+    var parsed = url.parse(link);
+    if (/^[\u0000-\u007f]*$/.test(parsed.path)) {
```

Comment:
According to my reading of https://en.wikipedia.org/wiki/Uniform_Resource_Locator and https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Generic_syntax allowed characters are more restrictive than this range - and the overall behavior more tricky. What do you think about using something like https://www.npmjs.com/package/uri-js to handle the nuances of international URLs instead?